app/assets/javascripts/angular/controllers/pages/recipe_finder_controller.js.erb

```diff
@@ -4,53 +4,86 @@ app.controller('RecipeFinderCtrl', ['$scope', '$http', '$routeParams', function(
   $scope.search_parameters.categories = [];
   $scope.search_parameters.keywords = [];
   $scope.search_parameters.timeframe = [0,120];
+  $scope.maxSize = 5;
+  // paginate varibales
+  $scope.bigTotalItems = 1;
+  $scope.bigCurrentPage = 1;
+
+  {// functions for getting recipes and recipe count
+    var getInitRecipes = function(){
+      $http.get('/recipes.json').success(function(data, status, xhr){
+        $scope.recipes = data.recipes;
+      });
+    }
 
+    var getInitRecipesCount = function(){
+      $http.get('/recipes_count.json').success(function(data, status, xhr){
+        $scope.bigTotalItems = data.count;
+      });
+    }
 
-  // submit search parameters and return a new list of recipes
-  $scope.submitParams = function(){
-    var data = { 
-      "search_parameters": $scope.search_parameters
+    var getFilteredRecipes = function(data){
```

Comment:
data contains user search_params and page number that we want to go to -> if we don't have data, the system will return "all" the recipes instead of recipes that the user filtered